### PR TITLE
Improve template macros for grub command line

### PR DESCRIPTION
**Pull request by teacup-on-rockingchair**
#### Description:

- Some border cases of strange state of `/etc/default/grub` file revealed some minor issues with current remediations

#### Rationale:

-  Cover the case where default grub file has the GRUB_CMDLINE_LINUX line but it is commented or the case where no such line exists

### CODE REVIEW

**Comment by jan-cerny** [comment]
@teacup-on-rockingchair It would be great to add a test scenario that would test this situation.

File: linux_os/guide/system/network/network-ipv6/disabling_ipv6/grub2_ipv6_disable_argument/tests/cmd_line_commented_etcdefaultgrub.fail.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bas
+# platform = Red Hat Enterprise Linux 7,sle12,sle15
```
Comment (jan-cerny) [inline linux_os/guide/system/network/network-ipv6/disabling_ipv6/grub2_ipv6_disable_argument/tests/cmd_line_commented_etcdefaultgrub.fail.sh:2]: You can't use the short IDs, you should use full names:

```
# platform = Red Hat Enterprise Linux 7,Oracle Linux 7,SUSE Linux Enterprise 12,SUSE Linux Enterprise 15
```

File: linux_os/guide/system/network/network-ipv6/disabling_ipv6/grub2_ipv6_disable_argument/tests/cmd_line_not_there_etcdefaultgrub.fail.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bas
+# platform = Red Hat Enterprise Linux 7,sle12,sle15
```
Comment (jan-cerny) [inline linux_os/guide/system/network/network-ipv6/disabling_ipv6/grub2_ipv6_disable_argument/tests/cmd_line_not_there_etcdefaultgrub.fail.sh:2]: You can't use the short IDs, you should use full names

**Review by jan-cerny — CHANGES_REQUESTED** [review]

File: linux_os/guide/system/network/network-ipv6/disabling_ipv6/grub2_ipv6_disable_argument/tests/cmd_line_commented_etcdefaultgrub.fail.sh (hunk 1 above)
Comment (teacup-on-rockingchair) [reply linux_os/guide/system/network/network-ipv6/disabling_ipv6/grub2_ipv6_disable_argument/tests/cmd_line_commented_etcdefaultgrub.fail.sh]: used a multi_platform_sle instead , anyways that was the idea

**Review by jan-cerny — APPROVED** [review]
When the tests are executed on a virtual machine backend, they're OK:

```
[jcerny@fedora scap-security-guide{pr/10989}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel7 --datastream build/ssg-rhel7-ds.xml grub2_ipv6_disable_argument
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-08-21-0946/test_suite.log
WARNING - Script arg_not_there_grubenv.fail.sh is not applicable on given platform
WARNING - Script wrong_value_entries.fail.sh is not applicable on given platform
WARNING - Script wrong_value.fail.sh is not applicable on given platform
WARNING - Script invalid_rescue.pass.sh is not applicable on given platform
WARNING - Script arg_not_there_rhel8.fail.sh is not applicable on given platform
WARNING - Script correct_grubenv.pass.sh is not applicable on given platform
WARNING - Script wrong_value_rhel8.fail.sh is not applicable on given platform
WARNING - Script arg_not_there_rhel9.fail.sh is not applicable on given platform
WARNING - Script wrong_value_rhel9.fail.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_grub2_ipv6_disable_argument
INFO - Script arg_not_there_etcdefaultgrub.fail.sh using profile (all) OK
INFO - Script arg_not_there_etcdefaultgrub_recovery_disabled.fail.sh using profile (all) OK
INFO - Script arg_not_there_rhel7.fail.sh using profile (all) OK
INFO - Script correct_grubby.pass.sh using profile (all) OK
INFO - Script correct_recovery_disabled.pass.sh using profile (all) OK
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script wrong_value_etcdefaultgrub.fail.sh using profile (all) OK
INFO - Script wrong_value_etcdefaultgrub_recovery_disabled.fail.sh using profile (all) OK
INFO - Script wrong_value_rhel7.fail.sh using profile (all) OK
INFO - Script cmd_line_commented_etcdefaultgrub.fail.sh using profile (all) OK
INFO - Script cmd_line_not_there_etcdefaultgrub.fail.sh using profile (all) OK
[jcerny@fedora scap-security-guide{pr/10989}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel7 --remediate-using ansible --datastream build/ssg-rhel7-ds.xml grub2_ipv6_disable_argument
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-08-21-0957/test_suite.log
WARNING - Script arg_not_there_grubenv.fail.sh is not applicable on given platform
WARNING - Script invalid_rescue.pass.sh is not applicable on given platform
WARNING - Script wrong_value_entries.fail.sh is not applicable on given platform
WARNING - Script wrong_value.fail.sh is not applicable on given platform
WARNING - Script arg_not_there_rhel8.fail.sh is not applicable on given platform
WARNING - Script correct_grubenv.pass.sh is not applicable on given platform
WARNING - Script wrong_value_rhel8.fail.sh is not applicable on given platform
WARNING - Script arg_not_there_rhel9.fail.sh is not applicable on given platform
WARNING - Script wrong_value_rhel9.fail.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_grub2_ipv6_disable_argument
INFO - Script arg_not_there_etcdefaultgrub.fail.sh using profile (all) OK
INFO - Script arg_not_there_etcdefaultgrub_recovery_disabled.fail.sh using profile (all) OK
INFO - Script arg_not_there_rhel7.fail.sh using profile (all) OK
INFO - Script correct_grubby.pass.sh using profile (all) OK
INFO - Script correct_recovery_disabled.pass.sh using profile (all) OK
INFO - Script correct_value.pass.sh using profile (all) OK
INFO - Script wrong_value_etcdefaultgrub.fail.sh using profile (all) OK
INFO - Script wrong_value_etcdefaultgrub_recovery_disabled.fail.sh using profile (all) OK
INFO - Script wrong_value_rhel7.fail.sh using profile (all) OK
INFO - Script cmd_line_commented_etcdefaultgrub.fail.sh using profile (all) OK
INFO - Script cmd_line_not_there_etcdefaultgrub.fail.sh using profile (all) OK
```

**Comment by jan-cerny** [comment]
The CI fail on Rawhide is caused by missing dnf, the fail of testing farm on c8s is tracked in https://github.com/ComplianceAsCode/content/issues/10978, both fails aren't caused by the changes in this pull request.

**Pull request merged**